service/frontend/workflow_handler.go

```diff
@@ -3756,32 +3757,291 @@ func (wh *WorkflowHandler) GetWorkerBuildIdOrdering(ctx context.Context, request
 	return matchingResponse.Response, err
 }
 
+func (wh *WorkflowHandler) StartBatchOperation(
+	ctx context.Context,
+	request *workflowservice.StartBatchOperationRequest,
+) (_ *workflowservice.StartBatchOperationResponse, retError error) {
+	defer log.CapturePanic(wh.logger, &retError)
+
+	if wh.isStopped() {
+		return nil, errShuttingDown
+	}
+
+	if err := wh.versionChecker.ClientSupported(ctx, wh.config.EnableClientVersionCheck()); err != nil {
+		return nil, err
+	}
+
+	if request == nil {
+		return nil, errRequestNotSet
+	}
+
+	if len(request.GetJobId()) == 0 {
+		return nil, errBatchJobIDNotSet
+	}
```

Comment:
I feel like require caller to specify JobId is weird. I understand we need a unique ID to do dedup but usually caller sets RequestID to do dedup.